packages/trace-viewer/src/ui/modelUtil.ts

```diff
@@ -356,6 +356,10 @@ export function prevInList(action: ActionTraceEvent): ActionTraceEvent {
   return (action as any)[prevInListSymbol];
 }
 
+export function pageForAction(action: ActionTraceEvent): PageEntry {
+  return context(action).pages[0]; // TODO: figure out what to do about multiple pages.
```

Comment:
We should mark each action with a page, similarly how we mark them with `contextSymbol`.

Reply:
I'm a little unsure if I found the right solution for this. Do you mean like in https://github.com/microsoft/playwright/pull/32248/commits/9fbf0ded351aea84ffcc1e44978673b36d5e0d2c?

Reply:
implemented what we discussed over video in https://github.com/microsoft/playwright/pull/32248/commits/03723b3e99bcbf61008e635a59918d737d39325f